executor/src/witgen/jit/block_machine_processor.rs

```diff
@@ -0,0 +1,304 @@
+#![allow(unused)]
+use std::{
+    collections::{BTreeSet, HashSet},
+    fmt::Display,
+};
+
+use bit_vec::BitVec;
+use powdr_ast::analyzed::{AlgebraicReference, Identity};
+use powdr_number::FieldElement;
+
+use crate::witgen::{
+    evaluators::fixed_evaluator, jit::affine_symbolic_expression::AffineSymbolicExpression,
+    machines::MachineParts, FixedData,
+};
+
+use super::{
+    affine_symbolic_expression::Effect,
+    variable::{Cell, Variable},
+    witgen_inference::{FixedEvaluator, WitgenInference},
+};
+
+/// A processor for generating JIT code for a block machine.
+struct BlockMachineProcessor<'a, T: FieldElement> {
+    fixed_data: &'a FixedData<'a, T>,
+    machine_parts: MachineParts<'a, T>,
+    block_size: usize,
+    latch_row: usize,
+}
+
+impl<'a, T: FieldElement> BlockMachineProcessor<'a, T> {
+    /// Generates the JIT code for a given combination of connection and known arguments.
+    /// Fails if it cannot solve for the outputs, or if any sub-machine calls cannot be completed.
+    pub fn generate_code(
+        &self,
+        identity_id: u64,
+        known_args: BitVec,
+    ) -> Result<Vec<Effect<T, Variable>>, String> {
+        let connection = self.machine_parts.connections[&identity_id];
```

Comment:
Could also just store `connection.right`, we don't need anything from the LHS, do we?